be/src/util/file_utils.cpp

```diff
@@ -33,36 +33,57 @@
 
 #include <openssl/md5.h>
 
+#include "gutil/strings/substitute.h"
+
 #include "env/env.h"
 #include "olap/file_helper.h"
 #include "util/defer_op.h"
 
 namespace doris {
 
-Status FileUtils::create_dir(const std::string& dir_path) {
-    try {
-        if (boost::filesystem::exists(dir_path.c_str())) {
-            // No need to create one
-            if (!boost::filesystem::is_directory(dir_path.c_str())) {
-                std::stringstream ss;
-                ss << "Path(" << dir_path << ") already exists, but not a directory.";
-                return Status::InternalError(ss.str());
-            }
-        } else {
-            if (!boost::filesystem::create_directories(dir_path.c_str())) {
-                std::stringstream ss;
-                ss << "make directory failed. path=" << dir_path;
-                return Status::InternalError(ss.str());
+using strings::Substitute;
+
+Status FileUtils::create_dir(const std::string& path, Env* env) {
+    if (path.empty()) {
+        return Status::InvalidArgument(Substitute("Unknown primitive type($0)", path));
+    }
+
+    boost::filesystem::path p(path);
+
+    string partial_path;
+    for (boost::filesystem::path::iterator it = p.begin(); it != p.end(); ++it) {
+        partial_path = partial_path.empty() ? it->string() : partial_path + "/" + it->string();
+        bool is_dir = false;
+        
+        Status s = env->is_directory(partial_path, &is_dir);
+        
+        if (s.ok()) {
+            if (is_dir) {
+                // It's a normal directory.
+                continue;
             }
+
+            // Maybe a file or a symlink. Let's try to follow the symlink.
+            string real_partial_path;
+            RETURN_IF_ERROR(env->canonicalize(partial_path, &real_partial_path));
+            
+            s = env->is_directory(real_partial_path, &is_dir);
+            if (s.ok() && is_dir) {
+                // It's a symlink to a directory.
+                continue;
+            } 
```

Comment:
If this path is normal file, we should return ERROR. No need to call another create_dir